### PR TITLE
Adds explicit sorting of fields and messages

### PR DESCRIPTION
**Pull request by gzigzigzeo**

### CODE REVIEW

File: config/main.go [hunk 1]
```diff
@@ -350,6 +356,19 @@ func setConfigModeBlockFields(f []string) {
 	}
 }
 
+// setDurationType sets the custom duration type
```
[the remark this answers is not in the document]
Comment (gzigzigzeo) [reply config/main.go]: Fixed

File: config/main.go [hunk 2]
```diff
@@ -350,6 +356,19 @@ func setConfigModeBlockFields(f []string) {
...
+func setSort(arg string) {
+	a := strings.ToLower(trimArg(arg))
+	if a == "" {
+		return
+	}
+
+	if a == "1" || a == "true" || a == "yes" {
```
[the remark this answers is not in the document]
Comment (gzigzigzeo) [reply config/main.go]: Thank you, completely forgot about those parsers, fixed.

File: config/main.go [hunk 3]
```diff
@@ -350,6 +357,27 @@ func setConfigModeBlockFields(f []string) {
...
+		log.Printf("Sorting is disabled, invalid value: %v. Use 1 or true to enable, 0 or false to disable.", a)
+		return
+	}
+
+	if b {
+		Sort = true
+		log.Printf("Sorting is enabled")
+	} else {
+		log.Printf("Sorting is disabled")
+	}
```
Comment (Joerger) [inline config/main.go:378]: It looks like `setSort` can be called twice if both `yaml` config is set and `params` are passed to `config.Read()`. In other `set` functions, the `params` input overwrites the `yaml` config, but this isn't the case for `setSort` if `params.sort=false` and `yaml.sort=true`. I would just change this to match the logic of other `set` functions.
```suggestion
	Sort = b
	
	log.Printf("Sorting enabled: %v", Sort)
```

File: config/main.go [hunk 4]
```diff
@@ -91,6 +92,9 @@ var (
 	// FieldNameReplacements represents map of CamelCased field names to under_score field names if needs replacement
 	FieldNameReplacements map[string]string = make(map[string]string)
 
+	// Sort sort fields and messages by name (otherwise, will keep the order as it was in .proto file)
+	Sort bool = false
```
Comment (Joerger) [inline config/main.go:96]: nit: bool is false by default
```suggestion
	Sort bool
```

File: config/main.go [hunk 5]
```diff
@@ -350,6 +357,27 @@ func setConfigModeBlockFields(f []string) {
...
+// setSort sets the custom duration type
+func setSort(arg string) {
+	a := strings.ToLower(trimArg(arg))
+	if a == "" {
+		return
+	}
+
+	b, err := strconv.ParseBool(a)
+	if err != nil {
+		log.Printf("Sorting is disabled, invalid value: %v. Use 1 or true to enable, 0 or false to disable.", a)
```
Comment (Joerger) [inline config/main.go:369]:
```suggestion
		log.Printf("Invalid value for sort: %v. Use 1 or true to enable, 0 or false to disable.", a)
```

File: config/main.go (hunk 3 above)
Comment (gzigzigzeo) [reply config/main.go:378]: Good point, thank you. Fixed.

**Review by Joerger — APPROVED** [review]

**Pull request closed without merging**